src/main/java/com/mongodb/kafka/connect/source/schema/BsonDocumentToSchema.java

```diff
@@ -37,7 +38,8 @@ public final class BsonDocumentToSchema {
   public static final String DEFAULT_FIELD_NAME = "default";
   private static final Logger LOGGER = LoggerFactory.getLogger(BsonDocumentToSchema.class);
   private static final String ID_FIELD = "_id";
-  static final Schema INCOMPATIBLE_SCHEMA_TYPE = Schema.OPTIONAL_STRING_SCHEMA;
+  static final Schema INCOMPATIBLE_SCHEMA_TYPE =
```

Comment:
Is this now effectively serving as a sentinel and used with reference equality semantics?

Reply:
Correct - it allows logging of the whole array / document rather than just the lowest individual values.